### PR TITLE
 Dcat graphing

### PR DESCRIPTION
**Pull request by stefina**

### CODE REVIEW

**Comment by metaodi** [comment]
Why did you add a __init__.py at the root level?

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    'kriminalitat': ['crime', 'public-order'],
+    'gesundheit': ['health'],
+    'basiskarten': ['territory', 'geography'],
+    'energie': ['energy'],
+    'bildung': ['education'],
+    'bevolkerung': ['population'],
+    'bauen-und-wohnen': ['construction'],
+}
+mapping_rights_dict = {
+    'cc-by-sa': 'NonCommercialAllow - CommercialAllow - ReferenceRequired',
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Value should be `NonCommercialAllowed-CommercialAllowed-ReferenceRequired`

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    'cc-zero': 'NonCommercialAllow - CommercialAllow - ReferenceNotRequired',
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Value should be `NonCommercialAllowed-CommercialAllowed-ReferenceNotRequired`

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    'umwelt': ['territory'],
+    'politik': ['politics'],
+    'mobilitat': ['mobility'],
+    'kultur': ['culture'],
+    'kriminalitat': ['crime', 'public-order'],
+    'gesundheit': ['health'],
+    'basiskarten': ['territory', 'geography'],
+    'energie': ['energy'],
+    'bildung': ['education'],
+    'bevolkerung': ['population'],
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Key should be `bevoelkerung`

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+    'preise': ['prices'],
+    'tourismus': ['tourism'],
+    'volkswirtschaft': ['national-economy'],
+    'wirtschaft': ['national-economy'],
+    'verwaltung': ['administration'],
+    'soziales': ['social-security'],
+    'freizeit': ['territory'],
+    'umwelt': ['territory'],
+    'politik': ['politics'],
+    'mobilitat': ['mobility'],
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Key should be `mobilitaet`

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 5]
```diff
@@ -0,0 +1,272 @@
...
+    'kultur': ['culture'],
+    'kriminalitat': ['crime', 'public-order'],
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Key should probably be `kriminalitaet`, does not exist yet on the portal, but I strongly suggest to use it to be consistent

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+mapping_accrualPerdiodicty = {
+    'halbjaehrlich': 'semiannual',
+    'jaehrlich': 'annual',
+    'laufend': 'continuous',
+    'monatlich': 'monthly',
+    'quartalsweise': 'quarterly',
+    'stuendlich': 'continuous',
+    'taeglich': 'daily',
+    'woechentlich': 'weekly',
+}
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Use to values for dct:accrualPeriodicity from http://handbook.opendata.swiss/en/library/ch-dcat-ap, maybe even use `http://purl.org/cld/freq/completelyIrregular` if needed

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+
+        # Basic date fields
+        date_items = [
+            ('metadata_modified', DCT.modified, None),
+            ('metadata_created', DCT.issued, None),
+        ]
+        self._add_date_triples_from_dict(dataset_dict, dataset_node, date_items)
+
+        # Organization
+        organization_id = pylons.config.get('ckan.organization_id', None)
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Please use a more descriptive name for this config option and maybe use ckanext.stadtzh-theme as a prefix (we already have `ckanext.stadtzh-theme.descr_file`, I know not the best name either :wink: ). I'd suggest `ckanext.stadtzh-theme.dcat_ap_organization_slug`. WDYT?

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 8]
```diff
@@ -0,0 +1,272 @@
...
+        description = self._get_dataset_value(dataset_dict, 'notes')
+        g.add((dataset_node, DCT.identifier, Literal(id + '@' + organization_id)))
+        g.add((dataset_node, DCT.title, Literal(title, lang=ckan_locale_default)))
+        g.add((dataset_node, DCT.description, Literal(description, lang=ckan_locale_default)))
+
+        # Update Interval
+        update_interval = self._get_dataset_value(dataset_dict, 'updateInterval')
+        accrualPeriodicity = mapping_accrualPerdiodicty.get(update_interval[0])
+        if accrualPeriodicity:
+            g.add((dataset_node, DCT.accrualPeriodicity, Literal(accrualPeriodicity)))
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: This should be a DCT.Frequency not a Literal

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 9]
```diff
@@ -0,0 +1,272 @@
...
+            g.add((contact_details, VCARD.hasEmail, URIRef(maintainer_email)))
+
+            items = [
+                ('contact_name', VCARD.fn, ['maintainer', 'author']),
+            ]
+            self._add_triples_from_dict(dataset_dict, contact_details, items)
+
+        # Tags
+        for tag in dataset_dict.get('tags', []):
+            g.add((dataset_node, DCAT.keyword, Literal(tag['name'])))
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Shouldn't tags have a lang attribute?

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 10]
```diff
@@ -0,0 +1,272 @@
...
+                    g.add((distribution, DCT['format'],
+                           Literal(resource_dict['format'])))
+
+                if resource_dict.get('mimetype'):
+                    g.add((distribution, DCAT.mediaType,
+                           Literal(resource_dict['mimetype'])))
+
+            # URL
+            url = resource_dict.get('url')
+            download_url = resource_dict.get('download_url')
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: resource_dict never has a download_url field, this only exists in DCAT. To know if it's a download or not, you have to check the format. `xml`, `wms`, `wmts`, `wfs`, `rest` are services, and thus don't need a downloadURL field, everything else is a download, where downloadURL and accessURL is the same.

File: ckanext/stadtzhtheme/dcat/profiles.py [hunk 11]
```diff
@@ -0,0 +1,272 @@
...
+                               URIRef(resource_dict['hash_algorithm'])))
+                    else:
+                        g.add((checksum, SPDX.algorithm,
+                               Literal(resource_dict['hash_algorithm'])))
+                g.add((distribution, SPDX.checksum, checksum))
+
+        # Publisher
+        if dataset_dict.get('organization'):
+
+            publisher_name = dataset_dict['organization']['title']
```
Comment (metaodi) [inline ckanext/stadtzhtheme/dcat/profiles.py]: Use the `author` field here, on the frontend this is called "Datenowner"

File: ckanext/stadtzhtheme/dcat/profiles.py (hunk 5 above)
Comment (stefina) [inline ckanext/stadtzhtheme/dcat/profiles.py]: We changed the generation of group-names to use munge_title_by_name, which will then produce the names as stated above. Group-names with Umlaut like ä ö and ü will just be converted to a o and u (not ae, oe and ue). Let's discuss this later :)

**Pull request merged**